weed/server/volume_server_handlers.go

```diff
@@ -56,20 +57,33 @@ func (vs *VolumeServer) privateStoreHandler(w http.ResponseWriter, r *http.Reque
 		vs.guard.WhiteList(vs.DeleteHandler)(w, r)
 	case "PUT", "POST":
 
-		// wait until in flight data is less than the limit
 		contentLength := getContentLength(r)
-
+		startTime := time.Now()
+		vs.inFlightUploadDataLimitCond.L.Lock()
```

Comment:
too many locks. The lock scope is also too big.

why not use the original atomic.AddInt64 operations?

Reply:
modified